pkg/envoy/rds/response.go

```diff
@@ -56,5 +58,40 @@ func NewResponse(cataloger catalog.MeshCataloger, proxy *envoy.Proxy, _ *xds_dis
 		rdsResources = append(rdsResources, ingressRouteConfig)
 	}
 
+	if dr != nil {
+		// Ensure all RDS resources are responded to a given non-nil and non-empty request
+		// Empty RDS RouteConfig will be provided for resources requested that our logic did not fulfill
+		// due to policy logic
+		rdsResources = ensureRDSRequestCompletion(dr, rdsResources)
+	}
+
 	return rdsResources, nil
 }
+
+// ensureRDSRequestCompletion computes delta between requested resources and response resources.
+// If any resources requested were not responded to, this function will fill those in with empty RouteConfig stubs
+func ensureRDSRequestCompletion(dr *xds_discovery.DiscoveryRequest, rdsResources []types.Resource) []types.Resource {
+	requestMapset := mapset.NewSet()
+	for idx := range dr.ResourceNames {
+		requestMapset.Add(dr.ResourceNames[idx])
```

Comment:
This should work too, right? 
```suggestion
	for _, resourceName := range dr.ResourceNames {
		requestMapset.Add(resourceName)
```

Reply:
:ok_hand: